packages/dev/core/src/Compute/computeShader.ts

```diff
@@ -402,8 +416,12 @@ export class ComputeShader {
             }
         }
 
-        this._engine.computeDispatch(this._effect, this._context, this._bindings, x, y, z, this._options.bindingsMapping, this.gpuTimeInFrame);
-
+        if ("buffer" in options) {

```

Comment:
let s avoid the in or instanceof in this part of the code if possible ?

Reply:
What would be the best way of doing so?

I could, for example, directly call the internal `_computeDispatch(` method of the engine. In that case, I'd also have to tweak the `_computeDispatch(` signature to allow a `StorageBuffer` in addition to a `DataBuffer`.

Reply:
good question, let see what @Popov72 thinks ?